client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/Issue.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.client.api;
+
+import javax.annotation.CheckForNull;
+
+public interface Issue {
+
+  String getSeverity();
+
+  @CheckForNull
+  Integer getStartLine();
+
+  @CheckForNull
+  Integer getStartLineOffset();
+
+  @CheckForNull
+  Integer getEndLine();
+
+  @CheckForNull
+  Integer getEndLineOffset();
+
+  String getMessage();
+
+  String getRuleKey();
+
+  String getRuleName();
+
+  /**
+   * @return null for global issues
+   */
+  @CheckForNull
+  public ClientInputFile getInputFile();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) "public" is redundant in this context. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2333)